src/main/java/ch/uzh/ifi/fs22/sel/group15/cozycave/server/controller/ListingController.java

```diff
@@ -48,30 +44,169 @@ public class ListingController {
+                try {
+                    temp.add(Gender.valueOf(genderString.toUpperCase()));
+                } catch (IllegalArgumentException e) {
+                    throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "Gender for AVAILABLE_to does not exist");
+                }
+            }
+            filtersMap.put("AVAILABLE_TO", temp);
+        }
+        if (listingType.isPresent()) {
+            // for conversion from String to ENUM
+            List<ListingType> temp = new ArrayList<>();
+            for (String listingString : listingType.get()) {
+                try {
+                    temp.add(ListingType.valueOf(listingString.toUpperCase()));
+                } catch (IllegalArgumentException e) {
+                    throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "LISTING_TYPE does not exist");
+                }
+            }
+            filtersMap.put("LISTING_TYPE", temp);
+        }
+        if (city.isPresent()) filtersMap.put("CITY", city.get());
+        if (zipCode.isPresent()) filtersMap.put("ZIP_CODE", zipCode.get());
+        if (minSqm.isPresent()) {
+            try {
+                filtersMap.put("MIN_SQM", Integer.parseInt(minSqm.get()));
+            } catch (NumberFormatException e) {
+                throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "Expected a number");
+            }
+        }
+        if (maxSqm.isPresent()) {
+            try {
+                filtersMap.put("MAX_SQM", Integer.parseInt(maxSqm.get()));
+            } catch (NumberFormatException e) {
+                throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "Expected a number");
+            }
+        }
+
+        if (available.isPresent()) filtersMap.put("AVAILABLE", Boolean.parseBoolean(available.get()));
+
+        if (filtersMap.size() > 0) {
+            List<FilterPair> filterPairs = filtersMap.entrySet().stream()
+                    .map(e -> new FilterPair(ListingFilters.getFilter(e.getKey()), e.getValue())).toList();
+
+            allListings = ListingFilter.createFilter(allListings).filter(filterPairs).getListings();
+        }
+
+        if (sort.isPresent()) {
+
+            Sorting sorting;
+            try {
+                sorting = Sorting.valueOf(sort.get().toUpperCase());
+            } catch (IllegalArgumentException e) {
+                throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "Sorting not supported");
+            }
+
+            OrderType orderType = OrderType.ASC;
+
+            if (order.isPresent()) {
+                try {
+                    orderType = OrderType.valueOf(order.get().toUpperCase());
+                } catch (IllegalArgumentException e) {
+                    throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "OrderType not found, only ASC and DESC supported");
+                }
+            }
+
+            // set default ordering to ASC or use userinput, only works for rent at the moment
```

Comment:
then change this to `TODO:` and create an (bug)

Reply:
This will not be a todo, this is intentionally. According to what should you be able to sort if nothing is selected.

Reply:
but it says: "set default ordering to ASC or use userinput, only works for rent at the moment". doesn't "at the moment" mean that it should be changed later on? or am I misunderstanding the comment. it says that the default sorting is ASC on the rent and that, if available, another order can be used but only on rent, doesn't it? and I think you mean that the default is rent.ASC but rent.DESC, sqm.ASC and sqm.DESC is also possible, right?